internal/orchestrator/restore_access_control_additional_test.go

```
@@ -0,0 +1,1144 @@
+		t.Cleanup(func() { restoreFS = origFS })
+		fakeFS := NewFakeFS()
+		t.Cleanup(func() { _ = os.RemoveAll(fakeFS.Root) })
+		restoreFS = fakeFS
+
+		canceled, cancel := context.WithCancel(context.Background())
+		cancel()
+		if err := applyPVEAccessControlFromStage(canceled, newTestLogger(), "/stage"); !errors.Is(err, context.Canceled) {
+			t.Fatalf("expected context.Canceled, got %v", err)
+		}
+
+		if err := applyPVEAccessControlFromStage(nil, newTestLogger(), "/stage"); err != nil {
```

golangci-lint (2.12.2)

[error] SA1012: do not pass a nil Context, even if a function permits it; pass context.TODO if you are unsure about which Context to use

(staticcheck)